src/dymoprint/q_dymo_label_widgets.py

```diff
@@ -273,6 +256,105 @@ def render_label(self):
             return self.render_engine.render_empty()
+        
+
+class BarcodeWithTextDymoLabelWidget(BaseDymoLabelWidget):
+    """
+    A widget for rendering labels with barcode and text below it using the Dymo label printer.
+    Args:
+        render_engine (DymoRenderEngine): An instance of the DymoRenderEngine class.
+        parent (QWidget): The parent widget of this widget.
+    Attributes:
+        render_engine (DymoRenderEngine): An instance of the DymoRenderEngine class.
+        label (QLineEdit): A QLineEdit widget for entering the content of the
+            barcode label.
+        codding (QComboBox): A QComboBox widget for selecting the type of barcode
+            to render.
+        font_style (QComboBox): The font style selection dropdown.
+        font_size (QSpinBox): The font size selection spinner.
+        draw_frame (QSpinBox): The frame width selection spinner.
+    Signals:
+        content_changed(): Emitted when the content of the label or the selected
+            barcode type changes.
+    Methods:
+        __init__(self, render_engine, parent=None): Initializes the widget.
+        render_label(self): Renders the barcode label using the current content
+            and barcode type.
+    """
+
+    def __init__(self, render_engine, parent=None):
+        super().__init__(parent)
+        self.render_engine = render_engine
+
+        self.label = QPlainTextEdit("text")
+        self.label.setFixedHeight(15 * (len(self.label.toPlainText().splitlines()) + 2))
+        # self.label = QLineEdit("")
+        self.setFixedHeight(self.label.height() + 10)
+        self.font_style = QComboBox()
+        self.font_size = QSpinBox()
+        self.font_size.setMaximum(150)
+        self.font_size.setMinimum(0)
+        self.font_size.setSingleStep(1)
+        self.font_size.setValue(90)
+        self.draw_frame = QSpinBox()
+        self.align = QComboBox()
+
+        self.align.addItems(["left", "center", "right"])
+
+        for name, font_path in parse_fonts():
+            self.font_style.addItem(name, font_path)
+            if "Regular" in name:
+                self.font_style.setCurrentText(name)
+
+        layout = QHBoxLayout()
+        item_icon = QLabel()
+        item_icon.setPixmap(QIcon(str(ICON_DIR / "barcode_text_icon.png")).pixmap(32, 32))
+        item_icon.setAlignment(QtCore.Qt.AlignmentFlag.AlignHCenter)
+        self.codding = QComboBox()
+        self.codding.addItems(AVAILABLE_BARCODES)
+
+        layout.addWidget(item_icon)
+        layout.addWidget(self.label)
+        layout.addWidget(QLabel("Font:"))
+        layout.addWidget(self.font_style)
+        layout.addWidget(QLabel("Size [%]:"))
+        layout.addWidget(self.font_size)
+        layout.addWidget(QLabel("Frame Width:"))
+        layout.addWidget(self.draw_frame)
+        layout.addWidget(QLabel("Alignment:"))
+        layout.addWidget(self.align)
+        layout.addWidget(QLabel("Codding:"))
```

Comment:
"Codding" is not a word. I see you copied it from the other barcode type, so this should be corrected there as well. Ideally we also fix the `.codding` attributes as well, but that's less important.

I'd recommend changing the text from `"Codding:"` to `"Type:"` and the attribute from `.codding` to `.barcode_type`.